src/main/java/org/cirdles/topsoil/app/TSVTable.java

```diff
@@ -41,8 +41,8 @@
  */
 public class TSVTable extends TableView<Entry> {
 
-    private Path savePath;
-    private Dataset dataset;
+    private Optional<DatasetResource> datasetResource = Optional.empty();
+    private Optional<Dataset> unsavedDataset = Optional.empty(); //Due to pasteFromClipboard
```

Comment:
I don't think a `DatasetResource` necessarily has a path, hence my comments about some of the methods there. While some _subtypes_ of `DatasetResource` have paths, I think the basic `DatasetResource` can just do CRUD operations and know nothing about the filesystem.